pycontrails/datalib/ecmwf/era5_model_level.py

```diff
@@ -163,12 +159,10 @@ def __init__(
             raise ValueError(msg)
         self.product_type = product_type
 
-        if product_type == "reanalysis" and ensemble_members:
+        if product_type != "ensemble_members" and ensemble_members:
             msg = "No ensemble members available for reanalysis product type."
             raise ValueError(msg)
-        if product_type == "ensemble_members" and not ensemble_members:
-            ensemble_members = ALL_ENSEMBLE_MEMBERS
-        self.ensemble_members = ensemble_members
+        self.ensemble_members = ensemble_members or ALL_ENSEMBLE_MEMBERS
```

Comment:
I think either is reasonable. Like you figured out, I wanted `self.ensemble_members: list[int]` instead of `self.ensemble_members: None | list[int]`, but I'm happy to revert.

Reply:
a7a6589a2d00479d1d9f931d32cbdaae77c815fc